frontend/src/metabase/admin/settings/components/widgets/FormattingWidget.tsx

```diff
@@ -0,0 +1,291 @@
+              value={timeStyle}
+              inputType="radio"
+              options={
+                getTimeStyleOptions("default").map(({ name, value }) => ({
+                  label: name,
+                  value,
+                })) ?? []
+              }
+              onChange={(newValue) =>
+                handleChange({
+                  ...localValue,
+                  "type/Temporal": {
+                    ...localValue?.["type/Temporal"],
+                    time_style: newValue as string,
+                  },
+                })
+              }
+            />
+          </FormattingSection>
+          <Divider mt="md" mb="md" />
+          <FormattingSection title={t`Numbers`}>
+            <FormattingInput
+              id="number_separators"
+              label={t`Separator style`}
+              value={numberSeparators}
+              inputType="select"
+              options={[
+                { label: "100,000.00", value: ".," },
+                { label: "100 000,00", value: ", " },
+                { label: "100.000,00", value: ",." },
+                { label: "100000.00", value: "." },
+                { label: "100'000.00", value: ".'" },
+              ]}
+              onChange={(newValue) =>
+                handleChange({
+                  ...localValue,
+                  "type/Number": {
+                    ...localValue?.["type/Number"],
+                    number_separators: newValue as string,
+                  },
+                })
+              }
+            />
+          </FormattingSection>
+          <Divider mt="md" mb="md" />
+          <FormattingSection title={t`Currency`}>
+            <FormattingInput
+              id="currency"
+              label={t`Unit of currency`}
+              value={currency}
+              inputType="select"
+              options={currencyOptions.map(({ name, value }) => ({
+                label: name,
+                value,
+              }))}
+              onChange={(newValue) =>
+                handleChange({
+                  ...localValue,
+                  "type/Currency": {
+                    ...localValue?.["type/Currency"],
+                    currency: newValue as string,
+                  },
+                })
+              }
+            />
+            <FormattingInput
+              id="currency_style"
+              label={t`Currency label style`}
+              value={currencyStyle}
+              inputType="radio"
+              options={getCurrencyStyleOptions(currency)}
+              onChange={(newValue) =>
+                handleChange({
+                  ...localValue,
+                  "type/Currency": {
+                    ...localValue?.["type/Currency"],
+                    currency_style: newValue as string,
+                  },
+                })
+              }
+            />
+          </FormattingSection>
+        </Stack>
+      )}
+    </Stack>
+  );
+}
+
+function FormattingInput({
```

Comment:
I can't use `BasicAdminSettingInput` here because of the prop `name: EnterpriseSettingKey`. But these inputs are not for different settings, they're all a part of `custom-formatting`. 

It could be an option to loosen the type to `name: string` in `BasicAdminSettingInput`